lib/geckodriver.js

```diff
@@ -0,0 +1,194 @@
+// transpile:main
+import events from 'events';
+import _ from 'lodash';
+import { logger, util } from 'appium-support';
+import { JWProxy } from 'appium-base-driver';
+import { SubProcess } from 'teen_process';
+import { quote } from 'shell-quote';
+import { getGeckodriverForOs } from './utils';
+
+const log = logger.getLogger('Geckodriver');
+
+const LOG_DEBUG = 'debug';
+const DEFAULT_LOG_LEVEL = LOG_DEBUG;
+
+const DEFAULT_HOST = '127.0.0.1';
+const DEFAULT_PORT = 29146;
+
+const MIN_SUPPORTED_VERSION = '0.26.0';
+const FIREFOX_PACKAGE = 'org.mozilla.firefox';
+const FIREFOX_MAIN_ACTIVITY = 'org.mozilla.gecko.BrowserApp';
+
+class Geckodriver extends events.EventEmitter {
```

Comment:
If the particular driver is not inherited from `BaseDriver` then capabilities validation does not work as expected. @imurchie Do you have possible workarounds for this problem or we must enforce inheritance from BaseDriver?

Reply:
@mykola-mokhnach @imurchie I was following chromedriver as example, should we extend this from BaseDriver?

Reply:
I would say it makes sense, since BaseDriver implements some basic procedures like commands scheduling, timers, capabilities verification and many other useful things. Chomerdirver is not being used on its own, but rather as an addition to other Android drivers, which already have the mentioned functionality.